src/Meshes/trailMesh.ts

```diff
@@ -0,0 +1,156 @@
+					Math.sin(j * alpha) * this._diameter,
+					-this._length + i
+				);
+				normals.push(
+					Math.cos(j * alpha),
+					Math.sin(j * alpha),
+					0
+				);
+			}
+			let l: number = positions.length / 3 - 2 * this._sectionPolygonPointsCount;
+			for (let j: number = 0; j < this._sectionPolygonPointsCount - 1; j++) {
+				indices.push(
+					l + j,
+					l + j + this._sectionPolygonPointsCount,
+					l + j + this._sectionPolygonPointsCount + 1,
+				);
+				indices.push(
+					l + j,
+					l + j + this._sectionPolygonPointsCount + 1,
+					l + j + 1
+				);
+			}
+			indices.push(
+				l + this._sectionPolygonPointsCount - 1,
+				l + this._sectionPolygonPointsCount - 1 + this._sectionPolygonPointsCount,
+				l + this._sectionPolygonPointsCount,
+			);
+			indices.push(
+				l + this._sectionPolygonPointsCount - 1,
+				l + this._sectionPolygonPointsCount,
+				l
+			);
+		}
+
+		data.positions = positions;
+		data.normals = normals;
+		data.indices = indices;
+		data.applyToMesh(this, true);
+	}
+
+    /**
+     * Update trailing mesh geometry.
+     */
+	public update() {
+		let positions = this.getVerticesData(VertexBuffer.PositionKind);
+		let normals = this.getVerticesData(VertexBuffer.NormalKind);
+		if (positions && normals) {
+			for (let i: number = 3 * this._sectionPolygonPointsCount; i < positions.length; i++) {
+				positions[i - 3 * this._sectionPolygonPointsCount] = positions[i] - normals[i] / this._length * this._diameter;
+			}
+			for (let i: number = 3 * this._sectionPolygonPointsCount; i < normals.length; i++) {
+				normals[i - 3 * this._sectionPolygonPointsCount] = normals[i];
+			}
+			let l: number = positions.length - 3 * this._sectionPolygonPointsCount;
+			let alpha: number = 2 * Math.PI / this._sectionPolygonPointsCount;
+			for (let i: number = 0; i < this._sectionPolygonPointsCount; i++) {
+				this._sectionVectors[i].copyFromFloats(
+					Math.cos(i * alpha) * this._diameter,
+					Math.sin(i * alpha) * this._diameter,
+					0
+				);
+				this._sectionNormalVectors[i].copyFromFloats(
+					Math.cos(i * alpha),
+					Math.sin(i * alpha),
+					0
+				);
+				Vector3.TransformCoordinatesToRef(this._sectionVectors[i], this._generator.getWorldMatrix(), this._sectionVectors[i]);
+				Vector3.TransformNormalToRef(this._sectionNormalVectors[i], this._generator.getWorldMatrix(), this._sectionNormalVectors[i]);
+			}
+			for (let i: number = 0; i < this._sectionPolygonPointsCount; i++) {
+				positions[l + 3 * i] = this._sectionVectors[i].x;
+				positions[l + 3 * i + 1] = this._sectionVectors[i].y;
+				positions[l + 3 * i + 2] = this._sectionVectors[i].z;
+				normals[l + 3 * i] = this._sectionNormalVectors[i].x;
+				normals[l + 3 * i + 1] = this._sectionNormalVectors[i].y;
+				normals[l + 3 * i + 2] = this._sectionNormalVectors[i].z;
+			}
+			this.updateVerticesData(VertexBuffer.PositionKind, positions, true, false);
+			this.updateVerticesData(VertexBuffer.NormalKind, normals, true, false);
+		}
+	}
+}
```

Comment:
The update mechanism should be called automatically and potentially turned off with an option if user wants to call it manually (Our golden rule is simplicity first :))